### PR TITLE
Added function for scroll_to 

### LINKED ISSUE #2834: [REF-2196] Create `rx.scroll_to` special event
We should have a special event handler called `rx.scroll_to("id")` that will scroll to the element with the given id.

The current way to accomplish this is

```python
rx.call_script(f"document.getElementById('{id}').scrollIntoView();")
```

But there should be a simpler API that wraps this.

<sub>[REF-2196](https://linear.app/reflex-dev/issue/REF-2196/create-rxscroll-to-special-event)</sub>

### PR DESCRIPTION
**Pull request by Yummy-Yums**
### All Submissions:

- [ ] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [ ] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

fixes #2834

### CODE REVIEW

**Comment by Yummy-Yums** [comment]
close #2834

**Comment by Yummy-Yums** [comment]
when i run `python ./scripts/pyi_generator.py`  to generate the component (the CI build fails with that error as well )
 , i get the below error  

image

File: reflex/event.py [hunk 1]
```diff
@@ -567,6 +567,23 @@ def set_clipboard(content: str) -> EventSpec:
...
+    """Select the id of a html element for scrolling into view.
+
+    Args:
+        elem_id: the id of the element
+
+    Returns:
+        EventSpec: an event to scroll the page to the selected element
+
+    """
+    elem_id = "#" + elem_id
```
Comment (ElijahAhianyo) [inline reflex/event.py:580]: I don't think we need the preceeding `#` character. we should use the elem_id as provided

**Comment by masenf** [comment]
supersede by #2948

**Pull request closed without merging**